include/sparrow/record_batch.hpp

```diff
@@ -539,8 +617,19 @@ namespace sparrow
         : m_name(name)
         , m_metadata(std::move(metadata))
         , m_name_list(to_vector<name_type>(std::forward<NR>(names)))
-        , m_array_list(to_vector<array>(std::forward<CR>(columns)))
     {
+        m_array_list.reserve(std::ranges::size(columns));
+        for (auto& col : columns)
+        {
+            if constexpr (std::is_lvalue_reference_v<CR>)
+            {
+                m_array_list.emplace_back(col);
+            }
+            else
+            {
+                m_array_list.emplace_back(std::move(col));
+            }
+        }
```

Comment:
If we store a variant of vector instead of a vector of variant, I think this method can be simplified with forwarding (it might require smoe help regarding intermediate types).